### PR TITLE
FIX: Prefabs and missing default control scheme in the inspector view of playerinput (ISXB-818)

### PR DESCRIPTION
**Pull request by bmalrat**
### Description

Fixed Prefabs and missing default control scheme used by PlayerInput component are now correctly shown in the inspector [ISXB-818](https://issuetracker.unity3d.com/product/unity/issues/guid/ISXB-818).
Before the change previously defined PlayerInput's default control scheme that doesn't exist exist any more are not shown and produce some error when entering in playmode. Disabling/enabling component silently remove the lost control scheme.
Also Default Scheme, Auto-Switch,Default Map fields was not well displayed(no blue override color and no context actions) for prefabs.

### Changes made

Changed the PlayerInput's inspector the show the missing default control scheme with a red background and a <Not Found> suffix.
Capture d'écran 2024-05-21 093655
The missing one is always in second position after <Any> and others schemes are now sorted in the popup.

Changed the ways that Default Scheme, Auto-Switch,Default Map are displayed to support prefabs.
Capture d'écran 2024-05-24 160723

### Notes

_Please write down any additional notes, remove the section if not applicable._

### Checklist

Before review:

- [X] Changelog entry added.
    - Explains the change in `Changed`, `Fixed`, `Added` sections.
    - For API change contains an example snippet and/or migration example.
    - FogBugz ticket attached, example `([case %number%](https://issuetracker.unity3d.com/issues/...))`.
    - FogBugz is marked as "Resolved" with *next* release version correctly set.
- [ ] Tests added/changed, if applicable.
    - Functional tests `Area_CanDoX`, `Area_CanDoX_EvenIfYIsTheCase`, `Area_WhenIDoX_AndYHappens_ThisIsTheResult`.
    - Performance tests.
    - Integration tests.
- [ ] Docs for new/changed API's.
    - Xmldoc cross references are set correctly.
    - Added explanation how the API works.
    - Usage code examples added.
    - The manual is updated, if needed.

During merge:

- [ ] Commit message for squash-merge is prefixed with one of the list:
    - `NEW: ___`.
    - `FIX: ___`.
    - `DOCS: ___`.
    - `CHANGE: ___`.
    - `RELEASE: 1.1.0-preview.3`.

### CODE REVIEW

**Comment by Pauliusd01** [comment]
This seems to mess with scheme order even when there are no missing schemes in the list (In the video I compare your PR with the current develop)
https://github.com/Unity-Technologies/InputSystem/assets/54306142/2b904247-bee1-44cc-a3ae-3e43cbd17c9e

**Comment by bmalrat** [comment]
> This seems to mess with scheme order even when there are no missing schemes in the list (In the video I compare your PR with the current develop) https://github.com/Unity-Technologies/InputSystem/assets/54306142/2b904247-bee1-44cc-a3ae-3e43cbd17c9e

I sorted them alphabetically like mentioned in that comment   [////TODO: sort alphabetically](https://github.com/Unity-Technologies/InputSystem/pull/1932/files#diff-f4fd031cfea34be0cf8758e698e7e2a4e2c530f38a12c87ed74a951f17a59e93L493)
I can remove the sort and/or the sort the scheme editor windows

**Comment by Pauliusd01** [comment]
> > This seems to mess with scheme order even when there are no missing schemes in the list (In the video I compare your PR with the current develop) https://github.com/Unity-Technologies/InputSystem/assets/54306142/2b904247-bee1-44cc-a3ae-3e43cbd17c9e
> 
> I sorted them alphabetically like mentioned in that comment [////TODO: sort alphabetically](https://github.com/Unity-Technologies/InputSystem/pull/1932/files#diff-f4fd031cfea34be0cf8758e698e7e2a4e2c530f38a12c87ed74a951f17a59e93L493) I can remove the sort and/or the sort the scheme editor windows

In my opinion It should be the same list both between the player component and the actual actions window. So probably best to not do any extra sorting in the component at all, if a missing scheme is selected it will already be quite visible by the red highlight. But maybe Hakan thinks otherwise

**Comment by bmalrat** [comment]
> > > This seems to mess with scheme order even when there are no missing schemes in the list (In the video I compare your PR with the current develop) https://github.com/Unity-Technologies/InputSystem/assets/54306142/2b904247-bee1-44cc-a3ae-3e43cbd17c9e
> > 
> > 
> > I sorted them alphabetically like mentioned in that comment [////TODO: sort alphabetically](https://github.com/Unity-Technologies/InputSystem/pull/1932/files#diff-f4fd031cfea34be0cf8758e698e7e2a4e2c530f38a12c87ed74a951f17a59e93L493) I can remove the sort and/or the sort the scheme editor windows
> 
> In my opinion It should be the same list both between the player component and the actual actions window. So probably best to not do any extra sorting in the component at all, if a missing scheme is selected it will already be quite visible by the red highlight. But maybe Hakan thinks otherwise

I removed the sort. If @ekcoh prefer with it I will revert the last change

**Review by Pauliusd01 — APPROVED** [review]

File: Packages/com.unity.inputsystem/CHANGELOG.md [hunk 1]
```diff
@@ -15,6 +15,7 @@ however, it has to be formatted properly to pass verification tests.
 - Fixed an issue where a composite binding would not be consecutively triggered after ResetDevice() has been called from the associated action handler [ISXB-746](https://issuetracker.unity3d.com/product/unity/issues/guid/ISXB-746).
 - Fixed resource designation for "d_InputControl" icon to address CI failure.
 - Fixed an issue where a composite binding would not be consecutively triggered after disabling actions while there are action modifiers in progress [ISXB-505](https://issuetracker.unity3d.com/product/unity/issues/guid/ISXB-505).
+- Fixed Missing default control scheme used by PlayerInput component are now correctly shown in the inspector [ISXB-818](https://issuetracker.unity3d.com/product/unity/issues/guid/ISXB-818)
```
Comment (ekcoh) [inline Packages/com.unity.inputsystem/CHANGELOG.md]: Am I correct to assume the behavior with a missing control scheme is equivalent to having it set to Any (ignoring Inspector representation)?

Comment (ekcoh) [reply Packages/com.unity.inputsystem/CHANGELOG.md]: Seems like this is mainly targeting representation of control scheme when opening a project/asset with a previously configured control scheme that no longer exist, or e.g. deleting a control scheme.

**Review by ekcoh — APPROVED** [review]
LGTM, added some questions just to make sure I understand the change correctly.

**Comment by ekcoh** [comment]
I think we can skip sort. Consistency between different places representing the same thing makes sense.

**Comment by Pauliusd01** [comment]
Sorry, had to change the status. I took a look at the original bug report and noticed that action map specific fields do not dirty the prefab is this not what the user had trouble with? Are we truly fixing the issue properly here?

**Comment by Pauliusd01** [comment]
Notice how none of the action specific fields get the blue highlights next to them when they're changed, so if the user wants to override their prefab they can't. The only way is to go into prefab mode and change it there.

https://github.com/Unity-Technologies/InputSystem/assets/54306142/6b87efc1-b433-48b5-924c-20011d6d62bd

File: Packages/com.unity.inputsystem/CHANGELOG.md (hunk 1 above)
Comment (bmalrat) [reply Packages/com.unity.inputsystem/CHANGELOG.md]: > Am I correct to assume the behavior with a missing control scheme is equivalent to having it set to Any (ignoring Inspector representation)?

yes

**Comment by bmalrat** [comment]
> Sorry, had to change the status. I took a look at the original bug report and noticed that action map specific fields do not dirty the prefab is this not what the user had trouble with? Are we truly fixing the issue properly here?

thanks for the highlight

**Review by Pauliusd01 — APPROVED** [review]

**Review by ekcoh — APPROVED** [review]
LGTM

**Pull request merged**